### PR TITLE
Renewable - refactor Renewable as Interface

### PR DESCRIPTION
**Pull request by jlhughes**
Renewable - refactor as Interface
UserCapabilityJunction - cleanup Created/LastModified Aware Interface

Refactored crunch Renewable to Interface so that Capability could implement it and replace all the fields I added the other day.  Also ended tidying up the User Created/LastModified interfaces as well as used by the UserCapabilityJunction. We had legacy code no longer required with our changes to Interface - Joel

### CODE REVIEW

**Comment by kgrgreer** [comment]
LGTM

**Review by noodlemoodle — APPROVED** [review]

File: src/foam/nanos/crunch/Capability.js [hunk 1]
```diff
@@ -157,46 +158,6 @@ foam.CLASS({
       class: 'String',
       includeInDigest: true,
     },
-    {
-      name: 'expiry',
```
Comment (noodlemoodle) [inline src/foam/nanos/crunch/Capability.js:161]: this shouldn't be removed since the expiry and etc is used to set the expiry on ucjs

Comment (noodlemoodle) [reply src/foam/nanos/crunch/Capability.js:161]: in `ConfigureUCJExpiryOnGranted`

Comment (jlhughes) [reply src/foam/nanos/crunch/Capability.js:161]: 'expiry' is provided by Renewable, which Capability now implements

Comment (noodlemoodle) [reply src/foam/nanos/crunch/Capability.js:161]: oops, yes, lgtm then

**Review by lchanmann — APPROVED** [review]

**Pull request merged**